sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1571,6 +1619,97 @@ a normal SYCL command-group submission.
 associated with the graph node resulting from this command-group submission is
 different from the one with which the dynamic_parameter was created.
 
+|===
+
+=== Dynamic Events [[dynamic-events]]
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+  class dynamic_event {
+    dynamic_event();
+    dynamic_event(const event& syclEvent);
+
+    void update(const event& syclEvent);
+  };
+}
+----
+
+Dynamic events represent SYCL events from outside of a given `command_graph`
+(either obtained from normal SYCL operations or from another `command_graph`)
+that nodes in that graph may depend on. The `dynamic_event` object enables these
+dependent events to be updated between graph executions.
+
+Dynamic events can be used to add dependencies to a graph node in the same way
+that regular SYCL events can, by passing them as parameters to
+`handler::depends_on()` inside the CGF which represents the node.
+
+[source,c++]
+----
+// Obtain an event from a normal queue submission
+event OutsideEvent = queue.submit(...);
+
+// Create a dynamic event to wrap that event
+ext::oneapi::experimental::dynamic_event DynEvent {OutsideEvent};
+
+// Add a graph node which depends on that dynamic event
+Graph.add([&](handler& CGH){
+  CGH.depends_on(DynEvent);
+  CGH.parallel_for(...);
+});
+----
```

Comment:
can we update DynEvent in this case? for example,

// Obtain another event from a normal queue submission
event e2 = queue.submit(...);
// or
event e3= anotherGraph.get_event(oneNode);

// and then
dynEvent.update(e2 or e3);

execGraph = Graph.finalize({sycl_ext::property::graph::updatable});
execGraph.update(node_added_at_line_1656);

Reply:
Yes, although in this case you would not need to call update to have the update to `dynEvent` reflected in `execGraph`, since the update happened before the graph was finalized.

In the same way as with `dynamic_parameter`s, updates to a `dynamic_event` are immediately reflected in the node and by extension the modifiable graph that node is part of.

Reply:
I see, and `execGraph.update(node_added_at_line_1656);` is a must with below code?

// Obtain another event from a normal queue submission
event e2 = queue.submit(...);
// or
event e3= anotherGraph.get_event(oneNode);

execGraph = Graph.finalize({sycl_ext::property::graph::updatable});

// and then
dynEvent.update(e2 or e3);

execGraph.update(node_added_at_line_1656);